build/xtask/src/sizes.rs

```diff
@@ -0,0 +1,208 @@
+    } else if toml.target.starts_with("thumbv8m") {
+        armv8m_suggest
+    } else {
+        panic!("Unknown target: {}", toml.target);
+    };
+
+    let mut suggestions = Vec::new();
+    let mut check_task =
+        |name: &str, stacksize: u32, requires: &IndexMap<String, u32>| {
+            let mut elf_name = dist_dir.clone();
+            elf_name.push(name);
+            let buffer = std::fs::read(elf_name)?;
+            let elf = match Object::parse(&buffer)? {
+                Object::Elf(elf) => elf,
+                o => bail!("Invalid Object {:?}", o),
+            };
+
+            let mut memory_sizes = IndexMap::new();
+            for phdr in &elf.program_headers {
+                // The PhysAddr is always a flash address, meaning we use the
+                // "on-disk" size of this section.
+                if let Some(region) = output_region(phdr.p_paddr) {
+                    *memory_sizes.entry(region).or_default() += phdr.p_filesz;
+                }
+                // If the VirtAddr disagrees with the PhysAddr, then this is a
+                // section which is relocated into RAM, so we also accumulate
+                // its MemSiz.
+                if phdr.p_paddr != phdr.p_vaddr {
+                    let region = output_region(phdr.p_vaddr).unwrap();
+                    *memory_sizes.entry(region).or_default() += phdr.p_memsz;
+                }
+            }
+            *memory_sizes.entry("ram").or_default() += stacksize as u64;
+
+            if !only_suggest {
+                writeln!(out, "{}", name)?;
+            }
+            let mut my_suggestions = Vec::new();
+            let mut has_asterisk = false;
+            for (mem_name, used) in memory_sizes {
+                let asterisk = name == "kernel" && mem_name == "ram";
+                has_asterisk |= asterisk;
+                if let Some(&size) = requires.get(mem_name) {
+                    let percent = used * 100 / size as u64;
+                    if !only_suggest {
+                        write!(
+                            out,
+                            "  {:<6} {: >5} bytes",
+                            format!("{}:", mem_name),
+                            used,
+                        )?;
+                        let mut color = ColorSpec::new();
+                        color.set_fg(Some(if percent >= 50 {
+                            Color::Green
+                        } else if percent > 25 {
+                            Color::Yellow
+                        } else {
+                            Color::Red
+                        }));
+                        out.set_color(&color)?;
+                        write!(out, " ({}%)", percent,)?;
+                        out.reset()?;
+                        writeln!(out, "{}", if asterisk { "*" } else { "" })?;
+                    }
+                    let suggestion = suggest(used);
+                    if suggestion != size as u64 && !asterisk {
```

Comment:
Change this to `<`, or change the subtract at the end of the routine (line 195) -- otherwise if the region is _smaller_ than the script thinks it should be, it crashes unceremoniously.